x-pack/legacy/plugins/siem/public/pages/network/index.tsx

```diff
@@ -5,24 +5,28 @@
  */
 
 import React from 'react';
-import { Redirect, Route, Switch } from 'react-router-dom';
-
-import { pure } from 'recompose';
-
-import { NetworkComponentProps } from '../../components/link_to/redirect_to_network';
+import { Redirect, Route, Switch, RouteComponentProps } from 'react-router-dom';
 
 import { IPDetails } from './ip_details';
 import { Network } from './network';
 
 const networkPath = `/:pageName(network)`;
-export const NetworkContainer = pure<NetworkComponentProps>(({ match }) => (
-  <>
-    <Switch>
-      <Route strict exact path={networkPath} render={props => <Network {...props} />} />
-      <Route path={`${networkPath}/ip/:detailName`} render={props => <IPDetails {...props} />} />
-      <Redirect from="/network/" to="/network" />
-    </Switch>
-  </>
+
+type Props = Partial<RouteComponentProps<{}>> & { url: string };
+
+export const NetworkContainer = React.memo<Props>(() => (
+  <Switch>
+    <Route strict exact path={networkPath} render={() => <Network />} />
+    <Route
+      path={`${networkPath}/ip/:detailName`}
+      render={({
+        match: {
+          params: { detailName },
+        },
+      }) => <IPDetails detailName={detailName} />}
```

Comment:
Note: Removed the `...props` spreading as that carries all the extra properties we are not interested in here as well as above in the ``<Network />` which didn't need the additional properties.

By removing the additional props spreading and just pushing down the string of `detailName` this will make it so the container does not constantly re-render because of the `match` object re-creation from `react-router`